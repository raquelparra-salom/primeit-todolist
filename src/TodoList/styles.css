.default-message {
    text-align: center;
    font-style: italic;
    color: rgb(55, 0, 255);
}