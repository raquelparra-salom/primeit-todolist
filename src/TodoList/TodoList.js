import React, { useState } from 'react';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';
import './styles.css';

function TodoList({ data, itemSelected, handleClickSelected, handleDoubleClickRemove }) {
    if (data.length === 0) {
        return <div className='default-message'> No hay elementos </div>;
    }
    return (

        // <>
        //     <ul>
        //         {items.map(item => <li key={item}>{item}</li>)}
        //     </ul>
        // </>
        <List>

            {data.map((item, i) => <ListItem key={i} disablePadding>
                <ListItemButton selected={item === itemSelected} onClick={() => { handleClickSelected(item) }} onDoubleClick={() => { handleDoubleClickRemove(item) }} >
                    <ListItemText primary={item} />
                </ListItemButton>
            </ListItem>)}
        </List>

    );
}

export default TodoList;
