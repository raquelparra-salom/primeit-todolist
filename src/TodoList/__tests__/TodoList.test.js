import React from 'react';
import { render } from '@testing-library/react';
import TodoList from '../TodoList';

test('render TodoList Empty', () => {


    const handleClickSelected = () => { }
    const handleDoubleClickRemove = () => { }
  
    const { container } = render(
        <TodoList data={[]} itemSelected={null} handleClickSelected={handleClickSelected} handleDoubleClickRemove={handleDoubleClickRemove}/>);
    expect(container).toMatchSnapshot();
});


test('render TodoList with items', () => {


    const handleClickSelected = () => { }
    const handleDoubleClickRemove = () => { }
  
    const { container } = render(
        <TodoList data={["item1", "item2", "item3"]} itemSelected={null} handleClickSelected={handleClickSelected} handleDoubleClickRemove={handleDoubleClickRemove}/>);
    expect(container).toMatchSnapshot();
});