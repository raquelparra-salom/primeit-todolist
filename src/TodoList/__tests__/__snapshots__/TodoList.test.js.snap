// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TodoList Empty 1`] = `
<div>
  <div
    class="default-message"
  >
     No hay elementos 
  </div>
</div>
`;

exports[`render TodoList with items 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            item1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            item2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            item3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </li>
  </ul>
</div>
`;
